src/ir/module-splitting.cpp

```diff
@@ -517,6 +521,31 @@ void ModuleSplitter::moveSecondaryFunctions() {
   }
 }
 
+Name ModuleSplitter::getTrampoline(Name funcName) {
+  auto trampolineIt = trampolineMap.find(funcName);
+  if (trampolineIt != trampolineMap.end()) {
+    return trampolineIt->second;
+  }
+
+  Builder builder(primary);
+  auto* oldFunc = secondary.getFunction(funcName);
+  auto trampoline = Names::getValidFunctionName(
+    primary, std::string("trampoline_") + funcName.toString());
+  trampolineMap[funcName] = trampoline;
```

Comment:
To avoid the second map lookup, you can use `insert` with an empty name up front, then either return the preexisting value or populate the map entry at the returned iterator depending on whether the `insert` succeeded.